code/messaging/src/main/java/com/adobe/marketing/mobile/messaging/internal/MessagingFullscreenMessageDelegate.java

```diff
@@ -81,13 +80,10 @@ public boolean overrideUrlLoad(final FullscreenMessage fullscreenMessage, final
             return true;
         }
 
-        URI uri;
+        // need to html unescape ampersands present in the link
+        final String localUrlString = urlString.replace(MessagingConstants.QueryParameters.AMPERSAND_HTML_ENTITY, "&");
```

Comment:
We are only unescaping `&`, what if other html escaped entities are present in the URL containing js?